src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1809,11 +1809,18 @@ class NodeConfig(object):
             vnodes = False
             pbs.logmsg(pbs.EVENT_DEBUG4,
                        '%s: vnode_per_numa_node is disabled' % caller_name())
-        vnode_name = self.hostname
         pbs.logmsg(pbs.EVENT_DEBUG4, '%s: numa nodes: %s' %
                    (caller_name(), self.numa_nodes))
+        vnode_name = self.hostname
+        # In some cases the hostname and vnode name do not match
+        if vnode_name not in vnode_list:
+            vnkeys = vnode_list.keys()
+            if len(vnkeys) == 1:
+                vnode_name = vnkeys[0]
+            else:
+                raise ProcessingError('Could not identify local vnode')
```

Comment:
I don't understand why the length of the vnode_list has to be 1?  Why is it that if the length of vnode_list is not one, we can't identify the local vnode?

Reply:
If there were more than one in the list, which one would we choose? There's really no way to know which is the correct one. The create_vnodes() method is only called by the exechost_startup handler, so there *should* only be one vnode key in the dictionary. If there happens to be more than one, all we can do is raise an error.

Reply:
Is this a point before you add more to the dictionary?  If there is only ever going to be one, why make it a dictionary?

Reply:
That's how it's passed to the hook, and that's how it gets passed back. It's part of the hooks infrastructure. Am I missing your point?

Reply:
Oh, I understand.  It's coming from the hook infrastructure before we create vnodes.  If there is more than one vnode, we have a problem.  Won't this fail if someone set up a vnodedef file?  Then we'll have more than one vnode.

Reply:
So *IF* we're on a system with a wonky hostname configuration *AND* the node has a vnodedef file with multiple vnodes, it will raise a ProcessingError. It's a "best effort" scenario.

Reply:
As long as we're not making the cgroups hook and vnode def files incompatible, I'm happy with this.